diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-document-picker/android/build.gradle node_modules/react-native-document-picker/android/build.gradle
--- node_modules.orig/react-native/react.gradle	2017-04-25 04:04:44.000000000 +0200
+++ node_modules/react-native/react.gradle	2018-09-15 02:02:30.943390900 +0200
@@ -125,6 +125,11 @@ afterEvaluate {
             from jsBundleDir
             into file(config."jsBundleDir${targetName}" ?:
                 "$buildDir/intermediates/assets/${targetPath}")
+                
+            // Workaround for Android Gradle Plugin 3.2+ new asset directory
+            from jsBundleDir
+            into file(config."jsBundleDir${targetName}" ?:
+                "$buildDir/intermediates/merged_assets/${targetPath}/merge${targetName}Assets/out")
 
             // mergeAssets must run first, as it clears the intermediates directory
             dependsOn(variant.mergeAssets)
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-document-picker/android/build.gradle node_modules/react-native-document-picker/android/build.gradle
--- node_modules.orig/react-native-document-picker/android/build.gradle	2017-04-25 04:04:44.000000000 +0200
+++ node_modules/react-native-document-picker/android/build.gradle	2018-09-15 02:02:30.943390900 +0200
@@ -1,21 +1,22 @@
 buildscript {
     repositories {
-        mavenCentral()
+        jcenter()
+        google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.2'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 'android-25'
-    buildToolsVersion '25.0.1'
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 22
+        targetSdkVersion 28
     }
 
     buildTypes {
@@ -26,16 +27,6 @@
     }
 }
 
-allprojects {
-    repositories {
-        mavenLocal()
-        jcenter()
-        maven {
-            url "http://files.couchbase.com/maven2/"
-        }
-    }
-}
-
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:0.57.1'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-fetch-blob/android/build.gradle node_modules/react-native-fetch-blob/android/build.gradle
--- node_modules.orig/react-native-fetch-blob/android/build.gradle	2017-08-03 03:35:14.000000000 +0200
+++ node_modules/react-native-fetch-blob/android/build.gradle	2018-09-15 02:25:53.718287300 +0200
@@ -1,26 +1,23 @@
 apply plugin: 'com.android.library'
 
-repositories {
-    mavenCentral()
-}
-
 buildscript {
     repositories {
         jcenter()
+        google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 23
+        targetSdkVersion 28
         versionCode 1
-        versionName "1.0"
+        versionName '1.0'
     }
     buildTypes {
         release {
@@ -28,11 +25,8 @@
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
-    productFlavors {
-    }
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
-    //{RNFetchBlob_PRE_0.28_DEPDENDENCY}
+    implementation 'com.facebook.react:react-native:0.57.1'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-file-viewer/android/build.gradle node_modules/react-native-file-viewer/android/build.gradle
--- node_modules.orig/react-native-file-viewer/android/build.gradle	2017-10-02 18:26:25.000000000 +0200
+++ node_modules/react-native-file-viewer/android/build.gradle	2018-09-15 02:25:54.141158200 +0200
@@ -1,25 +1,21 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 22
+        targetSdkVersion 28
         versionCode 1
-        versionName "1.0"
+        versionName '1.0'
     }
     lintOptions {
         abortOnError false
     }
 }
-
-repositories {
-    mavenCentral()
-}
-
+1
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:0.57.1'
 }
   
\ Kein Zeilenumbruch am Dateiende.
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-fs/android/build.gradle node_modules/react-native-fs/android/build.gradle
--- node_modules.orig/react-native-fs/android/build.gradle	1985-10-26 09:15:00.000000000 +0100
+++ node_modules/react-native-fs/android/build.gradle	2018-09-15 02:25:53.724271100 +0200
@@ -1,28 +1,25 @@
-def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-}
-
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.5.0'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion safeExtGet('compileSdkVersion', 26)
-    buildToolsVersion safeExtGet('buildToolsVersion', '26.0.3')
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
-        minSdkVersion safeExtGet('minSdkVersion', 16)
-        targetSdkVersion safeExtGet('targetSdkVersion', 22)
+        minSdkVersion  16
+        targetSdkVersion 28
         versionCode 1
-        versionName "1.0"
+        versionName '1.0'
     }
     lintOptions {
         abortOnError false
@@ -30,5 +27,5 @@
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:0.57.1'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-image-picker/android/build.gradle node_modules/react-native-image-picker/android/build.gradle
--- node_modules.orig/react-native-image-picker/android/build.gradle	2018-05-12 20:21:31.000000000 +0200
+++ node_modules/react-native-image-picker/android/build.gradle	2018-09-15 02:02:30.883579100 +0200
@@ -10,22 +10,23 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.+'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.2"
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 25
+        targetSdkVersion 28
         versionCode 1
         versionName computeVersionName()
     }
@@ -45,15 +46,15 @@
 }
 
 dependencies {
-    compile "com.facebook.react:react-native:+"  // From node_modules
-    
-    testCompile "junit:junit:4.10"
-    testCompile "org.assertj:assertj-core:1.7.0"
-    testCompile "org.robolectric:robolectric:3.3.2"
-
-    testCompile "org.easytesting:fest-assert-core:${FEST_ASSERT_CORE_VERSION}"
-    testCompile "org.powermock:powermock-api-mockito:${POWERMOCK_VERSION}"
-    testCompile "org.powermock:powermock-module-junit4-rule:${POWERMOCK_VERSION}"
-    testCompile "org.powermock:powermock-classloading-xstream:${POWERMOCK_VERSION}"
-    testCompile "org.mockito:mockito-core:${MOCKITO_CORE_VERSION}"
+    implementation "com.facebook.react:react-native:0.57.1"  // From node_modules
+
+    testImplementation "junit:junit:4.12"
+    testImplementation "org.assertj:assertj-core:2.6.0"
+    testImplementation "org.robolectric:robolectric:3.3.2"
+
+    testImplementation "org.easytesting:fest-assert-core:${FEST_ASSERT_CORE_VERSION}"
+    testImplementation "org.powermock:powermock-api-mockito:${POWERMOCK_VERSION}"
+    testImplementation "org.powermock:powermock-module-junit4-rule:${POWERMOCK_VERSION}"
+    testImplementation "org.powermock:powermock-classloading-xstream:${POWERMOCK_VERSION}"
+    testImplementation "org.mockito:mockito-core:${MOCKITO_CORE_VERSION}"
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-image-resizer/android/build.gradle node_modules/react-native-image-resizer/android/build.gradle
--- node_modules.orig/react-native-image-resizer/android/build.gradle	2017-05-08 16:36:22.000000000 +0200
+++ node_modules/react-native-image-resizer/android/build.gradle	2018-09-15 02:25:53.610070000 +0200
@@ -1,25 +1,26 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 23
+        targetSdkVersion 28
         versionCode 1
-        versionName "1.0"
+        versionName '1.0'
     }
     buildTypes {
         release {
@@ -28,10 +29,6 @@
     }
 }
 
-repositories {
-    mavenCentral()
-}
-
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:0.57.1'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-push-notification/android/build.gradle node_modules/react-native-push-notification/android/build.gradle
--- node_modules.orig/react-native-push-notification/android/build.gradle	2017-12-02 21:02:06.000000000 +0100
+++ node_modules/react-native-push-notification/android/build.gradle	2018-09-15 02:27:15.004800500 +0200
@@ -1,29 +1,23 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.3'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
-
-allprojects {
-    repositories {
-        jcenter()
-    }
-}
-
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 23
+        targetSdkVersion 28
         versionCode 1
-        versionName "1.0"
+        versionName '1.0'
         ndk {
             abiFilters "armeabi-v7a", "x86"
         }
@@ -37,10 +31,9 @@
 }
 
 dependencies {
-    compile fileTree(dir: 'libs', include: ['*.jar'])
-    testCompile 'junit:junit:4.12'
-    compile 'com.android.support:appcompat-v7:23.1.1'
-    compile 'com.facebook.react:react-native:+'
-    compile 'com.google.android.gms:play-services-gcm:+'
-    compile 'me.leolin:ShortcutBadger:1.1.8@aar'
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    testImplementation 'junit:junit:4.12'
+    implementation 'com.facebook.react:react-native:0.57.1'
+    implementation 'com.google.android.gms:play-services-gcm:15.0.1'
+    implementation 'me.leolin:ShortcutBadger:1.1.22@aar'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-securerandom/android/build.gradle node_modules/react-native-securerandom/android/build.gradle
--- node_modules.orig/react-native-securerandom/android/build.gradle	2017-10-11 19:39:28.000000000 +0200
+++ node_modules/react-native-securerandom/android/build.gradle	2018-09-15 02:25:53.944683200 +0200
@@ -1,34 +1,31 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.3.1'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "25.0.0"
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 22
+        targetSdkVersion 28
         versionCode 1
-        versionName "1.0"
+        versionName '1.0'
     }
     lintOptions {
         abortOnError false
     }
 }
 
-repositories {
-    mavenCentral()
-}
-
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:0.57.1'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-share-extension/android/build.gradle node_modules/react-native-share-extension/android/build.gradle
--- node_modules.orig/react-native-share-extension/android/build.gradle	2016-06-10 20:52:50.000000000 +0200
+++ node_modules/react-native-share-extension/android/build.gradle	2018-09-15 02:25:54.011505100 +0200
@@ -1,34 +1,31 @@
 buildscript {
   repositories {
     jcenter()
+    google()
   }
 
   dependencies {
-    classpath 'com.android.tools.build:gradle:1.1.3'
+    classpath 'com.android.tools.build:gradle:3.1.4'
   }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-  compileSdkVersion 23
-  buildToolsVersion "23.0.1"
+  compileSdkVersion 28
+  buildToolsVersion '28.0.2'
 
   defaultConfig {
     minSdkVersion 16
-    targetSdkVersion 22
+    targetSdkVersion 28
     versionCode 1
-    versionName "1.0"
+    versionName '1.0'
   }
   lintOptions {
     abortOnError false
   }
 }
 
-repositories {
-  mavenCentral()
-}
-
 dependencies {
-  compile "com.facebook.react:react-native:+"
+  implementation 'com.facebook.react:react-native:0.57.1'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-sqlite-storage/src/android/build.gradle node_modules/react-native-sqlite-storage/src/android/build.gradle
--- node_modules.orig/react-native-sqlite-storage/src/android/build.gradle	2018-05-28 14:28:26.000000000 +0200
+++ node_modules/react-native-sqlite-storage/src/android/build.gradle	2018-09-15 02:25:54.114229400 +0200
@@ -1,34 +1,31 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.0'
+        classpath 'com.android.tools.build:gradle:3.1.4'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion 28
+    buildToolsVersion '28.0.2'
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 22
+        targetSdkVersion 28
         versionCode 1
-        versionName "1.0"
+        versionName '1.0'
     }
     lintOptions {
         abortOnError false
     }
 }
 
-repositories {
-    mavenCentral()
-}
-
 dependencies {
-    compile 'com.facebook.react:react-native:0.14.+'
+    implementation 'com.facebook.react:react-native:0.57.1'
 }
diff -ruN -x '*.json' -x '*.xml' -x '*.java' -x '*.properties' -x '*.txt' -x '*.png' -x '*.class' -x '*.prefs' -x '*.flat' -x merge-state -x '*.jar' -x .classpath -x .project -x '*.iml' -x '*.so' -x '*.aidl' -x '*.aar' -x '*.store' -x '*.zip' -x '*.ap_' node_modules.orig/react-native-vector-icons/android/build.gradle node_modules/react-native-vector-icons/android/build.gradle
--- node_modules.orig/react-native-vector-icons/android/build.gradle	2018-07-28 10:26:36.000000000 +0200
+++ node_modules/react-native-vector-icons/android/build.gradle	2018-09-15 02:25:54.050400000 +0200
@@ -1,39 +1,31 @@
 buildscript {
   repositories {
     jcenter()
+    google()
   }
 
   dependencies {
-    classpath 'com.android.tools.build:gradle:2.3.+'
+    classpath 'com.android.tools.build:gradle:3.1.4'
   }
 }
 
 apply plugin: 'com.android.library'
 
-def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-}
-
-
 android {
-  compileSdkVersion safeExtGet('compileSdkVersion', 26)
-  buildToolsVersion safeExtGet('buildToolsVersion', '26.0.3')
+  compileSdkVersion 28
+  buildToolsVersion '28.0.2'
 
   defaultConfig {
-    minSdkVersion safeExtGet('minSdkVersion', 16)
-    targetSdkVersion safeExtGet('targetSdkVersion', 26)
+    minSdkVersion 16
+    targetSdkVersion 28
     versionCode 1
-    versionName "1.0"
+    versionName '1.0'
   }
   lintOptions {
     abortOnError false
   }
 }
 
-repositories {
-  mavenCentral()
-}
-
 dependencies {
-  compile "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
+  implementation 'com.facebook.react:react-native:0.57.1'
 }
